storage/redis/datasource.go

```diff
@@ -93,51 +103,54 @@ func (ds *DataSource) Get(ctx context.Context, variables octosql.Variables, stre
 		return nil, nil, errors.Wrap(err, "couldn't get all keys from filter")
 	}
 
-	if len(keysWanted.keys) == 0 {
-		allKeys := ds.client.Scan(0, "*", 0)
-
-		return &EntireDatabaseStream{
-			client:     ds.client,
-			dbIterator: allKeys.Iterator(),
-			isDone:     false,
-			alias:      ds.alias,
-			keyName:    ds.dbKey,
-		}, execution.NewExecutionOutput(execution.NewZeroWatermarkGenerator()), nil
+	rs := &RecordStream{
+		stateStorage: ds.stateStorage,
+		streamID:     streamID,
+		client:       ds.client,
+		alias:        ds.alias,
+		keyName:      ds.dbKey,
+		batchSize:    ds.batchSize,
 	}
 
-	sliceKeys := make([]string, 0)
-	for k := range keysWanted.keys {
-		sliceKeys = append(sliceKeys, k)
+	if len(keysWanted.keys) == 0 { // EntireDatabaseStream
+		rs.isEntireDatabaseStream = true
+	} else { // KeySpecificStream
+		sliceKeys := make([]string, 0)
```

Comment:
I know this wasn't introduced in this PR, but it just struck me. Isn't it better to simply write 
`rs.keys = make([]string, 0)`
`for k := range keysWanted.keys {`
     `    rs.keys = append(rs.keys, k)`
`}`

Reply:
yea you're right, after refactors this looks kinda dumb :P